src/mir/optimizations.md

```diff
@@ -26,6 +26,50 @@ optimizes it, and returns the improved MIR.
 [defid]: https://rustc-dev-guide.rust-lang.org/appendix/glossary.html#def-id
 [steal]: https://rustc-dev-guide.rust-lang.org/mir/passes.html?highlight=steal#stealing
 
+## Quickstart for adding a new optimization
+
+1. Make a Rust source file in `src/test/mir-opt` that shows the code you want to
+   optimize. This should be kept simple, so avoid `println!` or other formatting
+   code if it's not necessary for the optimization. The reason for this is that
+   `println!`, `format!`, etc. generate a lot of MIR that can make it harder to
+   understand what the optimization does to the test.
+
+2. Run `./x.py test --bless src/test/mir-opt/<your-test>.rs` to generate a MIR
+   dump. Read [this README][mir-opt-test-readme] for instructions on how to dump
+   things.
+
+3. Commit the current working directory state.
+
+4. Implement a new optimization in [`compiler/rustc_mir/src/transform`].
+   The fastest and easiest way to do this is to
+
+   1. pick a small optimization (such as [`no_landing_pads`]) and copy it
+      to a new file,
+   2. add your optimization to one of the lists in the
+      [`run_optimization_passes()`] function,
+   3. and then start modifying the copied optimization.
+
+5. Rerun `./x.py test --bless src/test/mir-opt/<your-test>.rs` to regenerate the
+   MIR dumps. Look at the diffs to see if they are what you expect.
+
+6. Run `./x.py test src/test/ui` to see if your optimization broke anything.
```

Comment:
I removed `--bless` because if you use that flag the tests may pass even though something changed. Or did you intend to use `--bless` because it's expected that some things will change due to the optimization being applied?

Reply:
No, leaving out bless is the right choice.